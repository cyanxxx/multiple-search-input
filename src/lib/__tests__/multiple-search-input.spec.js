import "jest";
import { shallowMount, createLocalVue, mount } from "@vue/test-utils";
// 直接读取打完包后的组件
import { MultipleSearchInput } from "../../../dist/multipleSearchInput.min.js";
import { FormTagsPlugin } from 'bootstrap-vue'

describe("MultipleSearchInput.vue", () => {
  const localVue  = createLocalVue()
  localVue.use(FormTagsPlugin)
  it("MultipleSearchInput have the default value.", () => {
    const value = ["a"];
    const list = [
      {
        text: "Apple",
        value: "a",
      },
      {
        text: "Bear",
        value: "b",
      },
      {
        text: "Cat",
        value: "c",
      },
    ];
    const wrapper = shallowMount(MultipleSearchInput, {
      propsData: { value, list },
      localVue
    });
    expect(wrapper.vm.tags).toEqual(['Apple'])
    expect(wrapper.vm.tagsId).toEqual(['a'])
  });
  it("MultipleSearchInput have the default value and list is async.", () => {
    const value = ["a"];
    const list = [];
    const options = [
      {
        text: "Apple",
        value: "a",
      },
      {
        text: "Bear",
        value: "b",
      },
      {
        text: "Cat",
        value: "c",
      },
    ];
    const wrapper = shallowMount(MultipleSearchInput, {
      propsData: { value, options, list },
      localVue
    });
    expect(wrapper.vm.tags).toEqual(['Apple'])
    expect(wrapper.vm.tagsId).toEqual(['a'])
  });
  it("MultipleSearchInput have the async default value and options.", async () => {
    const value = ["a"];
    const list = [];
    const options = [];
    const wrapper = shallowMount(MultipleSearchInput, {
      propsData: { value, options, list },
      localVue
    });
    jest.useFakeTimers();
    setTimeout(() => {
      wrapper.setProps({
      options:  [
        {
          text: "Apple",
          value: "a",
        },
        {
          text: "Bear",
          value: "b",
        },
        {
          text: "Cat",
          value: "c",
        },
      ]
      })
    }, 1000)
    setTimeout(() => {
      wrapper.setProps({
      value: ["a"]
      })
    }, 1000)

    jest.runAllTimers()
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.tags).toEqual(['Apple'])
    expect(wrapper.vm.tagsId).toEqual(['a'])
  });
  // it("depend the text input to find list", async () => {
  //   const value = [];
  //   const list = [{
  //         text: "Apple",
  //         value: "a",
  //       },
  //       {
  //         text: "Bear",
  //         value: "b",
  //       },
  //       {
  //         text: "Cat",
  //         value: "c",
  //       }];
  //   const options = [];
  //   const wrapper = mount(MultipleSearchInput, {
  //     propsData: { value, list },
  //     localVue
  //   });
  //   // expect(wrapper.element).toMatchSnapshot()
  //   const $input = wrapper.find('input')
  //   await $input.setValue('Apple')
  //   await $input.trigger('input')
  //   await $input.trigger('blur')
  //   // await wrapper.vm.$nextTick()
  //   // expect(wrapper.element).toMatchSnapshot()
  //   expect($input.element.value).toBe('Apple')
  //   expect(wrapper.vm.tags).toEqual(['Apple'])
  //   // expect(wrapper.vm.tagsId).toEqual(['a'])
  //   // expect(wrapper.emitted()['fetch-data']).toBe('A')
  // });
});
