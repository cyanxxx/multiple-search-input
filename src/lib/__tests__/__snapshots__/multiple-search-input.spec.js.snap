// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo render as usual 1`] = `
<div
  id="app"
  style="padding: 10px;"
>
  <div>
    <p>
      远程搜索, 取决于输入的字符，去服务器拿匹配数据
    </p>
     
    <multiplesearchinput-stub
      list=""
      options=""
      value=""
    />
     
    <p>
      已选值： [], list: []
    </p>
  </div>
   
  <div>
    <p>
      远程搜索，原本已选值，有初定数据和服务器拿取的数据
    </p>
     
    <multiplesearchinput-stub
      list=""
      options="[object Object],[object Object]"
      value="a,b"
    />
     
    <p>
      已选值： [
  "a",
  "b"
], list: [], option: [
  {
    "text": "Apple",
    "value": "a"
  },
  {
    "text": "Bear",
    "value": "b"
  }
] 
    </p>
  </div>
   
  <div>
    <p>
      远程搜索，可以自由写值，不一定从下拉选择, 没有初定数据
    </p>
     
    <multiplesearchinput-stub
      canfreetext="true"
      list=""
      options=""
      value=""
    />
     
    <p>
      已选值： [], list: []
    </p>
  </div>
   
  <div>
    <p>
      远程拿value，下拉默认值（此时list，option功能一致），本地搜索
    </p>
     
    <multiplesearchinput-stub
      list=""
      options=""
      value=""
    />
     
    <p>
      已选值： [], list: []
    </p>
     
    <button>
      外部加值
    </button>
  </div>
   
  <div>
    <p>
      处理前缀后缀空白值， 会重新修改value的值
    </p>
     
    <multiplesearchinput-stub
      canfreetext="true"
      list=" a "
      options=""
      value=" a "
    />
     
    <p>
      已选值： [
  " a "
], list: [
  " a "
]
    </p>
  </div>
   
  <div>
    <p>
      value默认值，远程拿list到值本地搜索
    </p>
     
    <multiplesearchinput-stub
      list=""
      options=""
      value="a"
    />
     
    <p>
      已选值： [
  "a"
], list: []
    </p>
  </div>
   
  <div>
    <p>
      没有选值，disabled状态
    </p>
     
    <multiplesearchinput-stub
      disabled="true"
      list="[object Object],[object Object],[object Object]"
      options=""
      value=""
    />
  </div>
   
  <div>
    <p>
      远程拿value默认值，翻页管理, 使用最好加多一层抽象，来集中管理pageNum，busy, loadedAll, total
    </p>
     
    <multiplesearchinput-stub
      infinite="true"
      list="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      options=""
      value=""
    />
     
    <p>
      已选值： [], list: [
  {
    "text": "0",
    "value": "0"
  },
  {
    "text": "1",
    "value": "1"
  },
  {
    "text": "2",
    "value": "2"
  },
  {
    "text": "3",
    "value": "3"
  },
  {
    "text": "4",
    "value": "4"
  },
  {
    "text": "5",
    "value": "5"
  },
  {
    "text": "6",
    "value": "6"
  },
  {
    "text": "7",
    "value": "7"
  },
  {
    "text": "8",
    "value": "8"
  },
  {
    "text": "9",
    "value": "9"
  },
  {
    "text": "10",
    "value": "10"
  },
  {
    "text": "11",
    "value": "11"
  },
  {
    "text": "12",
    "value": "12"
  },
  {
    "text": "13",
    "value": "13"
  },
  {
    "text": "14",
    "value": "14"
  },
  {
    "text": "15",
    "value": "15"
  },
  {
    "text": "16",
    "value": "16"
  },
  {
    "text": "17",
    "value": "17"
  },
  {
    "text": "18",
    "value": "18"
  },
  {
    "text": "19",
    "value": "19"
  },
  {
    "text": "20",
    "value": "20"
  },
  {
    "text": "21",
    "value": "21"
  },
  {
    "text": "22",
    "value": "22"
  },
  {
    "text": "23",
    "value": "23"
  },
  {
    "text": "24",
    "value": "24"
  },
  {
    "text": "25",
    "value": "25"
  },
  {
    "text": "26",
    "value": "26"
  },
  {
    "text": "27",
    "value": "27"
  },
  {
    "text": "28",
    "value": "28"
  },
  {
    "text": "29",
    "value": "29"
  },
  {
    "text": "30",
    "value": "30"
  },
  {
    "text": "31",
    "value": "31"
  },
  {
    "text": "32",
    "value": "32"
  },
  {
    "text": "33",
    "value": "33"
  },
  {
    "text": "34",
    "value": "34"
  },
  {
    "text": "35",
    "value": "35"
  },
  {
    "text": "36",
    "value": "36"
  },
  {
    "text": "37",
    "value": "37"
  },
  {
    "text": "38",
    "value": "38"
  },
  {
    "text": "39",
    "value": "39"
  },
  {
    "text": "40",
    "value": "40"
  },
  {
    "text": "41",
    "value": "41"
  },
  {
    "text": "42",
    "value": "42"
  },
  {
    "text": "43",
    "value": "43"
  },
  {
    "text": "44",
    "value": "44"
  },
  {
    "text": "45",
    "value": "45"
  },
  {
    "text": "46",
    "value": "46"
  },
  {
    "text": "47",
    "value": "47"
  },
  {
    "text": "48",
    "value": "48"
  },
  {
    "text": "49",
    "value": "49"
  }
]
    </p>
     
    <code>
      
       listConfig = {
          page: 1,
          total: 0,
          perPage: 10,
          busy: false,
          loadedAll: false
        }
        详细参照lib/multiple-async-input
     
    </code>
  </div>
</div>
`;
